kernel/kernel-defaults/src/main/java/io/delta/kernel/defaults/internal/json/JsonUtils.java

```diff
@@ -0,0 +1,221 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.defaults.internal.json;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import io.delta.kernel.data.*;
+import io.delta.kernel.types.*;
+
+import static io.delta.kernel.internal.util.Preconditions.checkArgument;
+
+import io.delta.kernel.defaults.internal.data.DefaultJsonRow;
+
+/**
+ * Utilities method to serialize and deserialize {@link Row} objects with a limited set of data type
+ * values.
+ * <p>
+ * Following are the supported data types:
+ * {@code boolean}, {@code byte}, {@code short}, {@code int}, {@code long}, {@code float},
+ * {@code double}, {@code string}, {@code StructType} (containing any of the supported subtypes),
+ * {@code ArrayType}, {@code MapType} (only a map with string keys is supported).
+ * <p>
+ * At a high-level, the JSON serialization is similar to that of Jackson's {@link ObjectMapper}.
+ */
+public class JsonUtils {
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    static {
+        OBJECT_MAPPER.registerModule(
+                new SimpleModule().addSerializer(Row.class, new RowSerializer()));
+    }
+
+    private JsonUtils() {
+    }
+
+    /**
+     * Converts a {@link Row} to a single line JSON string.
+     *
+     * @param row the row to convert
+     * @return JSON string
+     */
+    public static String rowToJson(Row row) {
+        try {
+            return OBJECT_MAPPER.writeValueAsString(row);
+        } catch (JsonProcessingException ex) {
+            throw new RuntimeException("Could not serialize row object to JSON", ex);
+        }
+    }
+
+    /**
+     * Converts a JSON string to a {@link Row}.
+     *
+     * @param json   JSON string
+     * @param schema to read the JSON according the schema
+     * @return {@link Row} instance with given schema.
+     */
+    public static Row rowFromJson(String json, StructType schema) {
+        try {
+            final JsonNode jsonNode = OBJECT_MAPPER.readTree(json);
+            return new DefaultJsonRow((ObjectNode) jsonNode, schema);
+        } catch (JsonProcessingException ex) {
+            throw new RuntimeException(String.format("Could not parse JSON: %s", json), ex);
+        }
+    }
```

Comment:
Unused now right?

Reply:
Should we comment that it's test-only for now? This is just another weird mix of data types it gets confusing